src/Symfony/Bundle/FrameworkBundle/Command/MessageConsumeCommand.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Message\Asynchronous\ConsumedMessage;
+use Symfony\Component\Message\MessageBusInterface;
+use Symfony\Component\Message\MessageConsumerInterface;
+
+/**
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+class MessageConsumeCommand extends Command
+{
+    protected static $defaultName = 'message:consume';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setDefinition(array(
+                new InputArgument('consumer', InputArgument::REQUIRED, 'Name of the consumer'),
+                new InputOption('bus', 'b', InputOption::VALUE_REQUIRED, 'Name of the bus to dispatch the messages to', 'message_bus'),
+            ))
+            ->setDescription('Consume a message')
+            ->setHelp(<<<'EOF'
+The <info>%command.name%</info> command consume a message and dispatch it to the message bus.
+
+    %command.full_name% <consumer-service-name>
+
+EOF
+            )
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        /** @var ContainerInterface $container */
+        $container = $this->getApplication()->getKernel()->getContainer();
```

Comment:
Can we use service locators instead of container?

Reply:
@Koc the point here is that the consumer - i.e. the service name - can be given as an argument. Not sure how would the service locator plays with that (as it needs to know the dependencies when constructed AFAIK)

Reply:
Require tags on consumers

Reply:
Correct, we can require the receivers (note the name change) to be tagged, so we can use service locators 👍

Reply:
Actually, I don't see a real point of using service locators in here. It forces us to create a new compiler pass and have another set of tags... while just using the container (as in many other commands in the FrameworkBundle) is the simplest option.

Reply:
Auto-configuration would be nice if we also define a generic folder (such as `src/MessageReceiver`). Without, it feels a bit too much work for not much benefits, as I explained in the [previous comment](https://github.com/symfony/symfony/pull/24411#discussion_r164315592). *But* obviously, if the general feedback is that we need this, we can add it (even later hehe :))

Reply:
Fetching the receiver from the container has one big drawback: it must be public. I would say go for an autoconfigured tag + scoped receiver locator

Reply:
Fair enough, updated to use service locators 👍